pkg/controller/postgresql.go

```diff
@@ -457,3 +466,40 @@ func (c *Controller) postgresqlDelete(obj interface{}) {
 
 	c.queueClusterEvent(pg, nil, spec.EventDelete)
 }
+
+/*
+  Ensures the service account required by StatefulSets to create pods exists in a namespace before a PG cluster is created there so that a user does not have to deploy the account manually.
+
+  The operator does not sync these accounts after creation.
+*/
+func (c *Controller) createPodServiceAccounts(event spec.ClusterEvent) error {
+
+	namespace := event.NewSpec.GetNamespace()
+	podServiceAccountName := c.opConfig.PodServiceAccountName
+
+	if _, ok := namespacesWithPodServiceAccount.Load(namespace); ok {
+		c.logger.Infof(fmt.Sprintf("The required pod service account %v already exists in the namespace %v", podServiceAccountName, namespace))
+		return nil
+	}
+
+	_, err := c.KubeClient.ServiceAccounts(namespace).Get(podServiceAccountName, metav1.GetOptions{})
+
+	if err != nil {
```

Comment:
We need to make sure the error is corresponds to the object not found. Use the `k8util.ResourceNotFound(err)` function for this. Otherwise, return the error.